client/js/s3/request-signer.js

```diff
@@ -46,20 +46,23 @@ qq.s3.RequestSigner = function(o) {
         },
         credentialsProvider,
 
-        generateHeaders = function(signatureConstructor, signature, promise) {
+        generateHeaders = function(signatureConstructor, signature, promise, credentials) {
+            credentials || (credentials = options.signatureSpec.credentialsProvider.get());
```

Comment:
I like how this eliminates some of the boilerplate in this method, but I think the following is a bit more readable:
```javascript
var credentials = credentials || options.signatureSpec.credentialsProvider.get());
```